filenamify.js

```diff
@@ -35,7 +35,9 @@ export default function filenamify(string, options = {}) {
 	const allowedLength = typeof options.maxLength === 'number' ? options.maxLength : MAX_FILENAME_LENGTH;
 	if (string.length > allowedLength) {
 		const extensionIndex = string.lastIndexOf('.');
-		string = string.slice(0, Math.min(allowedLength, extensionIndex)) + string.slice(extensionIndex);
+		const filename = extensionIndex > -1 ? string.slice(0, extensionIndex) : string;
```

Comment:
It's more readable to check for `=== -1` as it cannot be another negative number.